kafka-connect/kafka-connect/src/main/java/org/apache/iceberg/connect/channel/Coordinator.java

```diff
@@ -206,7 +206,15 @@ private void commitToTable(
 
     String branch = config.tableConfig(tableIdentifier.toString()).commitBranch();
 
+    // Control topic partition offsets may include a subset of partition ids if there were no
+    // records for other partitions.  Merge the updated topic partitions with the last committed
+    // offsets.
     Map<Integer, Long> committedOffsets = lastCommittedOffsetsForTable(table, branch);
+    Map<Integer, Long> mergedOffsets =
+        Stream.of(committedOffsets, controlTopicOffsets)
+            .flatMap(map -> map.entrySet().stream())
+            .collect(Collectors.toMap(Map.Entry::getKey, Map.Entry::getValue, Long::max));
```

Comment:
[doubt] IIUC this will merge both maps and in case of matching keys take the max Val ? wondering this case:

Lets say my committed offset : 
[0, 100] [2, 250]
my control topic offset : 
[1, 150] [2, 200]

my output [0, 100] [1,150] [2, 250], but i think we did went back as we [2, 200] in control offset ? is this check flagged already somewhere ?